File: adventofcode/java/year2016/day19/Day19Test.java
package adventofcode.java.year2016.day19;

import java.io.FileReader;
import java.io.IOException;
import java.io.StringReader;

import org.junit.Test;
import org.junit.Assert;

public class Day19Test {
    @Test
    public void part1Example() {
        var input = new StringReader("5");
        var output = "3";
        var result = Day19.part1(input);
        Assert.assertEquals("no error", "", result.error);
        Assert.assertEquals("correct output", output, result.answer);
    }

    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day19/actual.in")) {
            var output = "1830117";
            var result = Day19.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Example() {
         var input = new StringReader("5");
         var output = "2";
         var result = Day19.part2(input);
         Assert.assertEquals("no error", "", result.error);
         Assert.assertEquals("correct output", output, result.answer);
     }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/data/year2016/day19/actual.in")) {
             var output = "1417887";
             var result = Day19.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}

File: adventofcode/cpp/year2019/day19/day19_test.cc
#include "adventofcode/cpp/year2019/day19/day19.h"

#include <fstream>
#include <sstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day19, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day19/actual.in");
  std::string output = "197";
  adventofcode::answer_t a = day19::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day19, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day19/actual.in");
  std::string output = "9181022";
  adventofcode::answer_t a = day19::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}
